File: 0056-merge-intervals/0056-merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals.sort((a,b)=>{
        return a[0]- b[0];
    })

    const res =[];

    res.push(intervals[0]);

    const n = intervals.length;
    for(let i=1;i<n;i++){
        let oldvaluefirst = res[res.length-1][0];
        let oldvaluesecond = res[res.length-1][1];

        if(intervals[i][0]>oldvaluesecond){
            res.push(intervals[i]);
        }
        else{
            res.pop();
            res.push([oldvaluefirst, Math.max(oldvaluesecond, intervals[i][1])]);
        }

    }
    return res;
    
};

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
    
public:
     static bool mycomparator(const vector<int>& a, const vector<int>& b) {
        return a[0] < b[0];  // Corrected comparison
    }
    vector<vector<int>> merge(vector<vector<int>>& intervals) {

        sort(intervals.begin(), intervals.end(), mycomparator);
        int n = intervals.size();
        vector<vector<int>> result;
        result.push_back(intervals[0]);

        for (int i = 1; i < n; i++) {
            int start = intervals[i][0];
            int end = intervals[i][1];
            if (start > result[result.size() - 1][1]) {
                result.push_back({start,end});

            }
            else{
                result[result.size()-1][1]= max(result[result.size()-1][1], end);
            }
        }
        return result;
    }
};